sql/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -207,6 +207,10 @@ message Expression {
       Time time = 26;
     }
 
+    // The data type information of the literal.
+    // This field is only required for the root literal message.
+    DataType data_type = 100;
```

Comment:
Is this an optional field? For simple literals, such as byte, int, we can actually infer the data type.

Reply:
Messages in protobuf are always optional. If the data_type field is not set, we fall back to inferring the data type or using the deprecated data type fields.